cli/tools/task.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::colors;
+use crate::config_file::ConfigFile;
+use crate::flags::Flags;
+use crate::flags::TaskFlags;
+use crate::proc_state::ProcState;
+use deno_core::anyhow::bail;
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+fn get_tasks_config(
+  maybe_config_file: Option<&ConfigFile>,
+) -> Result<HashMap<String, String>, AnyError> {
+  if let Some(config_file) = maybe_config_file {
+    let maybe_tasks_config = config_file.to_tasks_config()?;
+    if let Some(tasks_config) = maybe_tasks_config {
+      for key in tasks_config.keys() {
+        if key.is_empty() {
+          bail!("Configuration file task names cannot be empty");
+        } else if !key
+          .chars()
+          .all(|c| c.is_ascii_alphanumeric() || matches!(c, '_' | '-'))
+        {
+          bail!("Configuration file task names must only contain alpha-numeric characters, underscores (_), or dashes (-). Task: {}", key);
+        } else if !key.chars().next().unwrap().is_ascii_alphabetic() {
+          bail!("Configuration file task names must start with an alphabetic character. Task: {}", key);
+        }
+      }
+      Ok(tasks_config)
+    } else {
+      bail!("No tasks found in configuration file")
+    }
+  } else {
+    bail!("No config file found")
+  }
+}
+
+fn print_available_tasks(tasks_config: HashMap<String, String>) {
+  eprintln!("{}", colors::green("Available tasks:"));
+
+  let mut task_names: Vec<String> = tasks_config.clone().into_keys().collect();
+  task_names.sort();
```

Comment:
I think we should leave tasks in the order they appear in the file. They might have some logical top to bottom flow between tasks and sorting it might break that flow. Thoughts?

Reply:
I was thinking the same. OTOH it's easier to scan for something when they are ordered, but I guess it's better to leave the order as users intended